src/io/flutter/logging/FlutterLog.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.logging;
+
+import com.intellij.execution.process.ProcessAdapter;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.util.Disposer;
+import com.intellij.openapi.util.Key;
+import com.jetbrains.lang.dart.ide.runner.server.vmService.VmServiceConsumers;
+import io.flutter.utils.VmServiceListenerAdapter;
+import org.dartlang.vm.service.VmService;
+import org.dartlang.vm.service.element.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FlutterLog {
+
+  // (Temporary) flag to enable logging support.
+  public static final boolean LOGGING_ENABLED = false;
+  public static final String LOGGING_STREAM_ID = "_Logging";
+
+  private static final FlutterLog INSTANCE = new FlutterLog();
+
+  private final List<Listener> listeners = new ArrayList<>();
+
+  // Shared instance.
+  private FlutterLog() {
+  }
+
+  public static FlutterLog getInstance() {
+    return INSTANCE;
+  }
+
+  public void addListener(@NotNull Listener listener, @Nullable Disposable parent) {
+    listeners.add(listener);
+    if (parent != null) {
+      Disposer.register(parent, () -> listeners.remove(listener));
+    }
+  }
+
+  public void removeListener(@NotNull Listener listener) {
+    listeners.remove(listener);
+  }
+
+  private void onEntry(@Nullable FlutterLogEntry entry) {
+    if (entry != null) {
+      for (Listener listener : listeners) {
+        listener.onEvent(entry);
+      }
+    }
+  }
+
+  public void listenToProcess(@NotNull ProcessHandler processHandler, @NotNull Disposable parent) {
+    processHandler.addProcessListener(new ProcessAdapter() {
+      @Override
+      public void onTextAvailable(@NotNull ProcessEvent event, @NotNull Key outputType) {
+        onEntry(FlutterLogEntryParser.parseDaemonEvent(event, outputType));
+      }
+    }, parent);
+  }
+
+  public void listenToVm(@NotNull VmService vmService) {
+    // No-op if disabled.
+    if (!LOGGING_ENABLED) return;
+
+    vmService.addVmServiceListener(new VmServiceListenerAdapter() {
+      @Override
+      public void received(String streamId, Event event) {
+        onVmServiceReceived(streamId, event);
+      }
+
+      @Override
+      public void connectionClosed() {
+        onVmConnectionClosed();
+      }
+    });
+
+    // Listen for logging events (note: no way to unregister).
+    vmService.streamListen(LOGGING_STREAM_ID, VmServiceConsumers.EMPTY_SUCCESS_CONSUMER);
```

Comment:
If we refactor so that the singleton lives off the PerfService object, we can piggy back on the existing listener there.

Reply:
Some other easy events we could log are GC events and frame events.

Reply:
(the events we currently get from flutter include `Flutter.FrameworkInitialization`, `Flutter.FirstFrame`, `Flutter.Frame`)

Reply:
I'm not sure we want to piggy-back on the PerfService since I think whether it is running is tied to the the `HeapDisplay`.  But maybe it shouldn't be. Anyway, I've migrated the log to live in the app and have added a TODO to consider moving this bit too.

Reply:
(I think this isn't right actually. Will follow-up in another PR.)

Reply:
Yeah, I have no real opinion of `FlutterApp` vs. `PerService`; I was mostly interested in there being one per flutter app session (not one per Intellij instance). looks good either was -